### PR TITLE
Remove iop resource name check for operator reconcile

### LINKED ISSUE #29394: Ignore reconciling IstioOperator CR by annotation only
To keep backward compatibility, we added the logic: https://github.com/istio/istio/pull/29378 to ignore any CR name starting with installed-state, but actually we should just check by annotation only. Created this issue to track the work, will make the change after 1.9

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Since we have been using the annotation `install.istio.io/ignoreReconcile` for a long time now, and since the file name check is not 100% accurate, we should rely solely on this annotation for our needs, and remove the file name check.

Fix https://github.com/istio/istio/issues/29394

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by zirain — APPROVED** [review]

File: releasenotes/notes/44152.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: installation
+releaseNotes:
+  - |
+    **Fixed** operator skip reconcile for iop resources named starting with `installed-state`, and now it only relies on annotation `install.istio.io/ignoreReconcile`.
```
Comment (ericvn) [inline releasenotes/notes/44152.yaml:6]: Not sure if this sounds better.
```suggestion
    **Removed** operator skip reconcile for iop resources with names starting with `installed-state`. It now relies solely on the annotation `install.istio.io/ignoreReconcile`.
```

File: releasenotes/notes/44152.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    This won't affect the behavior of `istioctl install`.
```
[the remark this answers is not in the document]
Comment (hanxiaop) [reply releasenotes/notes/44152.yaml:9]: Done, thanks

File: releasenotes/notes/44152.yaml (hunk 1 above)
Comment (hanxiaop) [reply releasenotes/notes/44152.yaml]: Revised together with the change above, thanks.

**Review by ericvn — APPROVED** [review]
Release Notes LGTM

**Pull request merged**